rtapp_function/fpBench/fp.c

```diff
@@ -0,0 +1,245 @@
+
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+
+/*  The  program  may  be  run  with  Float defined as either float or
+    double.  With IEEE arithmetic, the same answers are generated  for
+    either floating point mode.  */
+
+#define Float    double            /* Floating point type used in FFT */
+
+#define Asize    256               /* Array edge size */
+
+#ifndef ITERATIONS                 /* Allow iterations to be supplied by -D */
+#define ITERATIONS 100000
+#endif
+int niter = ITERATIONS;            /* Iteration counter */
+
+#define Passes   50                /* Number of FFT/Inverse passes */
+#define Ntimes   (ITERATIONS / Passes) /* Number of outer loop passes */
+
+#define max(a, b) ((a) > (b) ? (a) : (b))
+#define min(a, b) ((a) <= (b) ? (a) : (b))
+
+#ifndef unix
+#ifndef WIN32
+extern char *farmalloc(long s);
+#define malloc(x)   farmalloc(x)
+#endif
+#define FWMODE   "wb"
+#else
+#define FWMODE   "w"
+#endif
+
+/*
+
+        Multi-dimensional fast Fourier transform
+
+        Adapted from Press et al., "Numerical Recipes in C".
+
+*/
```

Comment:
Did you write this yourself or is the code taken from some other source?

Reply:
This is an floating point benchmark obtained from some external source. It was uploaded with the intent of showcasing that complex code can be ran as a rt-app phase. This probably should not be kept as part of the PR.

Reply:
I see.

Reply:
removed fp benchmark.